diff --git a/tidal-link/src/hs/Sound/Tidal/Link.hs b/tidal-link/src/hs/Sound/Tidal/Link.hs
index 714432f7..c243e8ee 100644
--- a/tidal-link/src/hs/Sound/Tidal/Link.hs
+++ b/tidal-link/src/hs/Sound/Tidal/Link.hs
@@ -55,6 +55,8 @@ instance Storable SessionState where
 foreign import ccall "abl_link.h abl_link_create"
   create :: BPM -> IO AbletonLink
 
+newtype CBool = CBool Int
+
 foreign import ccall "abl_link.h abl_link_enable"
   abl_link_enable :: AbletonLink -> CBool -> IO ()
 
